python/ray/data/_internal/execution/operators/actor_pool_map_operator.py

```diff
@@ -0,0 +1,374 @@
+import collections
+from typing import Dict, Any, Iterator, Callable, List, Tuple, Union, Optional
+
+import ray
+from ray.data.block import Block, BlockMetadata
+from ray.data.context import DatasetContext, DEFAULT_SCHEDULING_STRATEGY
+from ray.data._internal.execution.interfaces import (
+    RefBundle,
+    ExecutionResources,
+    ExecutionOptions,
+    PhysicalOperator,
+)
+from ray.data._internal.execution.operators.map_operator import (
+    MapOperator,
+    _map_task,
+    _TaskState,
+)
+from ray.types import ObjectRef
+from ray._raylet import ObjectRefGenerator
+
+
+class ActorPoolMapOperator(MapOperator):
+    """A MapOperator implementation that executes tasks on an actor pool."""
+
+    def __init__(
+        self,
+        transform_fn: Callable[[Iterator[Block]], Iterator[Block]],
+        input_op: PhysicalOperator,
+        name: str = "Map",
```

Comment:
"Map" -> "ActorPoolMap"?

Reply:
Good point! I'll change that for `TaskPoolMapOperator` as well.